Bug: ESLint config cli fails - eslint@9.3.0 vs 8.56.0

### Environment

Node version: v20.13.1
npm version: 10.5.2
Local ESLint version: 
Global ESLint version: 
Operating System: mcr.microsoft.com/devcontainers/typescript-node:1-20-bookworm


### What parser are you using?

Other

### What did you do?

<details>
<summary>Configuration</summary>

```
npm init @eslint/config@latest -- --debug
```
</details>

```js
<!-- Paste your code here -->
```


### What did you expect to happen?

Create config file and install dependencies.

### What actually happened?

✔ How would you like to use ESLint? · problems
✔ What type of modules does your project use? · commonjs
✔ Which framework does your project use? · none
✔ Does your project use TypeScript? · typescript
✔ Where does your code run? · browser
The config that you've selected requires the following dependencies:

eslint@9.x, globals, @eslint/js, typescript-eslint
✔ Would you like to install them now? · No / Yes
✔ Which package manager do you want to use? · npm
☕️Installing...
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: ts-common@1.0.0
npm ERR! Found: eslint@9.3.0
npm ERR! node_modules/eslint
npm ERR!   dev eslint@"9.x" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer eslint@"^8.56.0" from typescript-eslint@7.10.0
npm ERR! node_modules/typescript-eslint
npm ERR!   dev typescript-eslint@"*" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! 
npm ERR! For a full report see:
npm ERR! /home/node/.npm/_logs/2024-05-21T20_57_05_935Z-eresolve-report.txt

npm ERR! A complete log of this run can be found in: /home/node/.npm/_logs/2024-05-21T20_57_05_935Z-debug-0.log
A config file was generated, but the config file itself may not follow your linting rules.

### Link to Minimal Reproducible Example

https://github.com/diberry/eslint9-examples/tree/diberry/feature-examples/ts-common

### Participation

- [X] I am willing to submit a pull request for this issue.

### Additional comments

The config file was created.

--- a/lib/config-generator.js
+++ b/lib/config-generator.js
@@ -167,6 +167,7 @@ const compat = new FlatCompat({baseDirectory: __dirname, recommendedConfig: plug
 
         if (this.answers.language === "typescript") {
             this.result.devDependencies.push("typescript-eslint");
+            this.result.installFlags.push("--force"); // tseslint does not support eslint 9.x yet
             importContent += "import tseslint from \"typescript-eslint\";\n";
             exportContent += "  ...tseslint.configs.recommended,\n";
         }
@@ -182,7 +183,9 @@ const compat = new FlatCompat({baseDirectory: __dirname, recommendedConfig: plug
         if (this.answers.framework === "react") {
             if (this.answers.eslintVersion === "9.x") {
                 this.result.devDependencies.push("eslint-plugin-react", "@eslint/compat");
-                this.result.installFlags.push("--force");
+                if (!this.result.installFlags.includes("--force")) {
+                    this.result.installFlags.push("--force");
+                }
                 importContent += "import pluginReactConfig from \"eslint-plugin-react/configs/recommended.js\";\n";
                 importContent += "import { fixupConfigRules } from \"@eslint/compat\";\n";
                 exportContent += "  ...fixupConfigRules(pluginReactConfig),\n";

